src/java/test/org/apache/zookeeper/server/NIOServerCnxnTest.java

```diff
@@ -68,4 +74,38 @@ public void testOperationsAfterCnxnClose() throws IOException,
             zk.close();
         }
     }
+
+    /**
+     * Mock extension of NIOServerCnxn to test for
+     * CancelledKeyException (ZOOKEEPER-2044).
+     */
+    class MockNIOServerCnxn extends NIOServerCnxn {
+        public MockNIOServerCnxn( NIOServerCnxn cnxn )
+                throws IOException {
+            super( cnxn.zkServer, cnxn.sock, cnxn.sk, cnxn.factory );
```

Comment:
This line has extra spaces and I could see formatting issues in many places. Please remove unwanted spaces and correct it.